doc/hardware/porting/soc_porting.rst

```diff
@@ -0,0 +1,324 @@
+   drivers, sub-systems, boards, and other source code found in Zephyr.
+
+#. :file:`Kconfig.soc`: the base SoC configuration which defines a Kconfig SoC
+   symbol and provides the SoC name to the Kconfig ``SOC`` setting.
+   If the ``soc.yml`` describes a SoC family and series, then those must also
+   be defined in this file. Kconfig settings outside of the SoC tree must not be
+   selected. To select general Zephyr Kconfig settings the :file:`Kconfig` file
+   must be used.
+
+#. :file:`CMakeLists.txt`: CMake file loaded by the build system. This CMake
+   file can define additional include paths and/or source files to be used when
+   a build targets the SoC. Also the base line linker script to use must be
+   defined.
+
+The optional files are:
+
+- :file:`Kconfig`, :file:`Kconfig.defconfig` software configuration in
+  :ref:`kconfig` format. These select the architecture and peripherals
+  available.
+
+Write your SoC YAML
+*********************
+
+The SoC YAML file describes the SoC family, SoC series, and SoC at a high level.
+
+Detailed configurations, such as hardware description and configuration are done
+in devicetree and Kconfig.
+
+The skeleton of a simple SoC YAML file containing just one SoC is:
+
+.. code-block:: yaml
+
+   socs:
+   - name: <soc1>
+
+It is possible to have multiple SoC located in the SoC folder.
+For example if they belong to a common family or series it is recommended to
+locate such SoC in a common tree.
+Multiple SoCs and SoC series in a common folder can be described is the
+:file:`soc.yml` file as:
+
+.. code-block:: yaml
+
+   family:
+     name: <family-name>
+     series:
+     - name: <series-1-name>
+       socs:
+       - name: <soc1>
+         cpucluster:
+         - name: <coreA>
+         - name: <coreB>
+           ...
+       - name: <soc2>
+     - name: <series-2-name>
+       ...
+
+
+Write your devicetree
+*********************
+
+SoC devicetree include files are located in the :file:`<zephyr-repo>/dts` folder
+under a corresponding :file:`<ARCH>/<VENDOR>`.
+
+The SoC :file:`dts/<ARCH>/<VENDOR>/<soc>.dtsi` describes your SoC hardware in
+the Devicetree Source (DTS) format and must be included by any boards which use
+the SoC.
+
+If a highlevel :file:`<arch>.dtsi` file exists, then a good starting point is to
+include this file in your :file:`<soc>.dtsi`.
+
+In general, :file:`<soc>.dtsi` should look like this:
+
+.. code-block:: devicetree
+
+   #include <arch>/<arch>.dtsi
+
+   / {
+           chosen {
+                   /* common chosen settings for your SoC */
+           };
+
+           cpus {
+                   #address-cells = <m>;
+                   #size-cells = <n>;
+
+                   cpu@0 {
+                   device_type = "cpu";
+                   compatible = "<compatibles>";
+                   /* ... your CPU definitions ... */
+           };
+
+           soc {
+                   /* Your SoC definitions and peripherals */
```

Comment:
Should this clarify that this should only be for common resources for all cpus in that soc?

Reply:
we haven't done modification to dtsi as part of hwmv2 migration, so answer is that it depends a little on how the SoC developer decides to do it.

Several multi-core SoCs are still assuming that they are only included by the specific board dts file for a given CPU cluster identifier.

This makes it hard to set up rules, although we could add recommendations.
But before doing so, I think existing SoCs should follow such recommendations.